### PR TITLE
[ADD][10.0] add sale_product_set_variant

### PR DESCRIPTION
**Pull request by leemannd**
### Goal
Extension of the sale product set module to make it work with variants

### NB
This is WIP for the moment. I will change this description when ready

### CODE REVIEW

File: sale_product_set_variants/README.rst
Comment (simahawk) [inline sale_product_set_variants/README.rst]: can you complete this?

File: sale_product_set_variants/README.rst
Comment (simahawk) [inline sale_product_set_variants/README.rst:39]: update this list ;)

File: sale_product_set_variants/__manifest__.py
Comment (simahawk) [inline sale_product_set_variants/__manifest__.py]: ehm, copy pasted? :)

File: sale_product_set_variants/demo/product_set.xml
Comment (simahawk) [inline sale_product_set_variants/demo/product_set.xml:2]: you can merge this 2 demo files...

File: sale_product_set_variants/models/product_set_line.py
Comment (simahawk) [inline sale_product_set_variants/models/product_set_line.py]: be consistent w/ formatting: move the ')' to new line

File: sale_product_set_variants/README.rst
Comment (simahawk) [inline sale_product_set_variants/README.rst]: simplify this link, see https://github.com/OCA/maintainer-tools/tree/master/template/module

File: sale_product_set_variants/README.rst
Comment (simahawk) [reply sale_product_set_variants/README.rst]: while you do this, please update the screenshots which are pretty old and show them here.

File: sale_product_set_variants/tests/__init__.py
Comment (simahawk) [inline sale_product_set_variants/tests/__init__.py]: import this otherwise odoo's std tests won't run it

File: sale_product_set_variants/tests/test_product_set.py
Comment (simahawk) [inline sale_product_set_variants/tests/test_product_set.py]: better to use `assertAlmostEqual` in such cases

File: sale_product_set_variants/wizard/product_set_add.py
Comment (simahawk) [inline sale_product_set_variants/wizard/product_set_add.py]: same as above for ')'

File: sale_product_set_variants/wizard/product_set_add.py
Comment (simahawk) [inline sale_product_set_variants/wizard/product_set_add.py]: self.env.context.get('active_id')

File: sale_product_set_variants/wizard/product_set_add.py
Comment (simahawk) [inline sale_product_set_variants/wizard/product_set_add.py:86]: self.ensure_one()

File: sale_product_set_variants/wizard/product_set_add.py
Comment (simahawk) [inline sale_product_set_variants/wizard/product_set_add.py]: `if so_lines:` is enough

File: sale_product_set_variants/wizard/wizard_set_lines.py
Comment (simahawk) [inline sale_product_set_variants/wizard/wizard_set_lines.py]: respect the naming convention: `product.set.add.line`

File: sale_product_set_variants/wizard/wizard_set_lines.py
Comment (simahawk) [inline sale_product_set_variants/wizard/wizard_set_lines.py]: remove this line

File: sale_product_set_variants/wizard/wizard_set_lines.py
Comment (simahawk) [inline sale_product_set_variants/wizard/wizard_set_lines.py]: True

File: sale_product_set_variants/wizard/wizard_set_lines.py
Comment (simahawk) [reply sale_product_set_variants/wizard/wizard_set_lines.py]: also, theré's not much code here nor in the wizard file: merge them.

**Review by simahawk — CHANGES_REQUESTED** [review]

File: sale_product_set_variants/tests/test_product_set.py
Comment (TDu) [reply sale_product_set_variants/tests/test_product_set.py]: Could not this 2300.4 value be calculated ?
Instead of landing like a magical one.

**Comment by mpanarin** [comment]
@simahawk fixed ;)

File: sale_product_set_variants/__manifest__.py
Comment (simahawk) [inline sale_product_set_variants/__manifest__.py:2]: 2017 only

File: sale_product_set_variants/__manifest__.py
Comment (simahawk) [inline sale_product_set_variants/__manifest__.py:10]: sequence is for???

File: sale_product_set_variants/tests/test_product_set.py
Comment (simahawk) [inline sale_product_set_variants/tests/test_product_set.py:6]: Class name should be CamelCase

**Review by simahawk — CHANGES_REQUESTED** [review]

**Comment by simahawk** [comment]
Putting this on hold: I'm migrating to v11. I'm going to backport further fixes from there when ready.

**Comment by simahawk** [comment]
I'll get back on this only when #575 for v11 is merged

**Review by ghost — APPROVED** [review]

File: sale_product_set_variants/README.rst
Comment (SimoRubi) [inline sale_product_set_variants/README.rst:1]: Please use [this script](https://github.com/OCA/maintainer-tools/blob/master/tools/gen_addon_readme.py) to generate a nice README.

File: sale_product_set_variants/wizard/product_set_add.xml
Comment (SimoRubi) [inline sale_product_set_variants/wizard/product_set_add.xml:12]: the view gets messy when I try to change the variant:
image
try using the many2many widget

**Review by SimoRubi — CHANGES_REQUESTED** [review]
If this is ready to be reviewed, please update the description.

**Comment by SimoRubi** [comment]
@simahawk, https://github.com/OCA/sale-workflow/pull/575 has been merged

**Comment by leemannd** [comment]
@rafaelbn Tell me if you can merge it. If not, feel free to close it

**Comment by leemannd** [comment]
@simahawk AFAIR we don't need it anymore and I will not take any more time on this topic

**Comment by leemannd** [comment]
@simahawk @rafaelbn Closing it as old and outdated.

**Pull request closed without merging**
